tests/Javascript/Components/Tables/UsersBookingRequestTable.spec.js

```diff
@@ -0,0 +1,1442 @@
+          "pivot": {"role_id": 1, "permission_id": 20}
+        }]
+      }]
+    },
+    "reservations": [{
+      "id": 10,
+      "room_id": 3,
+      "booking_request_id": 10,
+      "start_time": "2021-04-20T17:53:57.000000Z",
+      "end_time": "2021-04-20T19:53:57.000000Z",
+      "created_at": "2021-03-28T17:52:30.000000Z",
+      "updated_at": "2021-03-28T17:52:30.000000Z",
+      "room": {
+        "id": 3,
+        "name": "Modular Conference Room",
+        "number": "H-711-4",
+        "floor": 7,
+        "building": "Hall",
+        "created_at": "2021-03-28T17:52:29.000000Z",
+        "updated_at": "2021-03-28T17:52:29.000000Z",
+        "status": "available",
+        "min_days_advance": 0,
+        "max_days_advance": 200,
+        "attributes": {
+          "food": false,
+          "sofas": 0,
+          "chairs": 10,
+          "tables": 1,
+          "alcohol": false,
+          "computer": false,
+          "projector": true,
+          "fundraiser": false,
+          "television": false,
+          "whiteboard": false,
+          "a_v_permitted": true,
+          "ambiant_music": false,
+          "coffee_tables": 0,
+          "sale_for_profit": false,
+          "capacity_sitting": "20",
+          "capacity_standing": "0"
+        },
+        "room_type": "Conference"
+      }
+    }]
+  }
+];
+
+jest.mock("axios");
+
+beforeEach(() => {
+  InertiaFormMock.error.mockClear()
+  InertiaFormMock.post.mockClear()
+
+  localVue = createLocalVue()
+  localVue.use(InertiaApp)
+  localVue.use(InertiaForm)
+
+});
+afterEach(() => {
+  localVue = null
+})
+
+test('should mount without crashing', () => {
+  const wrapper = shallowMount(BookingsTable, {
+    localVue,
+    propsData: {
+      bookings: bookTest
+    }
+  })
+})
+
+test('should set uniqueStatus array from array prop', () => {
+  const wrapper = shallowMount(BookingsTable, {
+    localVue,
+    propsData: {
+      bookings: bookTest
+    }
+  })
+
+  expect(wrapper.vm.uniqueStatuses).toEqual(
+    [ 'approved', 'review', 'refused']
+  )
+})
+
+test('should show advanced filters popup', () => {
+  const wrapper = shallowMount(BookingsTable, {
+    localVue,
+    propsData: {
+      bookings: bookTest
+    }
+  });
+  wrapper.vm.toggleAdvancedFilters();
+  expect(wrapper.vm.showFilterModal).toBe(true);
+})
+
+test('should clear json form date', () => {
+  const wrapper = shallowMount(BookingsTable, {
+    localVue,
+    propsData: {
+      bookings: bookTest
+    }
+  });
+  wrapper.vm.jsonForm.dateCheck = "2021 07 07";
+
+  wrapper.vm.clearDate();
+
+  expect(wrapper.vm.jsonForm.dateCheck).toBe(null);
+})
+
+test('should emit filterBookingsJson', () => {
+  const wrapper = shallowMount(BookingsTable, {
+    localVue,
+    propsData: {
+      bookings: bookTest
+    }
+  });
+
+  wrapper.vm.advancedFilters();
+
+  expect(wrapper.emitted().filterBookingsJson).toBeTruthy()
+})
+
+test('post sent to filterRooms route', () => {
+  const mocked =  {
```

Comment:
Could you move these large `consts` either outside of the tests or in a separate file? It will make the tests easier to read

Reply:
Moved